frontend/src/components/common/private-route/private-route.tsx

```diff
@@ -22,6 +24,17 @@ const PrivateRoute: FC<Props> = ({
     return <Navigate to={redirectTo} />;
   }
 
+  const requiredPermission = permissions.filter((permission) => {
+    return user?.permissions.some(
+      (userPermission) => userPermission === permission,
+    );
+  });
+  const hasRequiredPermission = Boolean(requiredPermission.length);
```

Comment:
```suggestion
const checkHasPermission = (pagePermissions: string[], userPermissions: string[]): boolean => {
  return pagePermissions.every((pagePermission) => {
    return userPermissions.some((userPermission) => userPermission=== pagePermission);
  });
};
```
add this helper to shared folder and use it here

```tsx
  const hasUser = Boolean(user);

  const hasPermission = checkHasPermission(permissions, user?.permissions ?? []);

  if (!hasUser || !hasPermission) {
    return <Navigate to={redirectTo} />;
  }
```